code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -313,6 +170,10 @@ var/list/gaslist_cache = init_gaslist_cache()
 	//Compares sample to self to see if within acceptable ranges that group processing may be enabled
 	//Returns: a string indicating what check failed, or "" if check passes
 
+/datum/gas_mixture/proc/react(turf/open/dump_location)

```

Comment:
This is how all these procs are defined in `gas_mixture.dm`. Defined empty and documented, and then after all the definitions they're given content. Apparently doing this actually​ marginally increases performance, and also makes it very easy to document them.